packages/react-composites/src/composites/CallComposite/pages/CallPage.tsx

```diff
@@ -53,10 +56,25 @@ export const CallPage = (props: CallPageProps): JSX.Element => {
   const errorBarProps = usePropsFor(ErrorBar);
   const mutedNotificationProps = useSelector(mutedNotificationSelector);
   const networkReconnectTileProps = useSelector(networkReconnectTileSelector);
+  const isLocalMicrophoneEnabled = useSelector(getLocalMicrophoneEnabled);
+  const adapter = useAdapter();
 
   // Reduce the controls shown when mobile view is enabled.
   const callControlOptions = mobileView ? reduceCallControlsForMobile(options?.callControls) : options?.callControls;
 
+  /**
+   * Since toggling microphone in lobby page doesn't actually change the microphone status in the call object,
+   * we track it as a ui state attribute `isLocalMicrophoneEnabled` and force the adapter to mute/unmute the call
```

Comment:
I worry about a small condition where:
* I elect unmute in the configuration page
* I mute myself in the lobby
* I join the call and there is a small time where I am unmuted in the call before `adapter.mute();` has finished
  * Or perhaps there is an error in muting me and suddenly I am unmuted in the call

Reply:
(the mitigation to this - probably leading to more spaghetti code - would be to always do joinCall({microphoneEnabled: false}), then when the call connects call adapter.unmute if the user was meant to be unmuted)

Reply:
Good point. However, the same condition can occur for unmuting the microphone (which I assume is safer).
This would require us to remove some code so I wouldn't worry about the increase in spaghetti code...

What do others think about this? @prprabhu-ms @PorterNan @alkwa-msft